### DISCUSSION TITLE
Pack invalid tar header

### DISCUSSION THREAD

**Discussion by ubermensch37** [buildpacks/community#204] [Q&A] [answered]
I'm getting the following error:

Inspecting buildpack blob: failed to get next tar entry: archive/tar: invalid tar header

for  cloudfoundry buildpack either cflinux3 or heroku builder:
ERROR: failed to build: downloading buildpack: extracting from https://github.com/cloudfoundry/java-buildpack.git: inspecting buildpack blob: failed to get next tar entry: archive/tar: invalid tar header

with heroku java buildpack and heroku builder:
ERROR: failed to build: downloading buildpack: extracting from https://github.com/cloudfoundry/java-buildpack.git: inspecting buildpack blob: failed to get next tar entry: archive/tar: invalid tar header

pack version
0.28.0+git-b10e41f.build-3729
macOS Catalina 10.15.7
go version go1.19.5 darwin/amd64

Can you help me to fix it?

**Comment by dfreilich** [comment, marked as answer]
Those buildpacks are from a different version of the technology. They are from Cloud Foundry's V2 Buildpacks, and are incompatible with tooling for Cloud Native Buildpacks. 

Their successor is the Paketo projects, built by former Cloud Foundry engineers: https://paketo.io/
You can also look into the Heroku and Google builders, as you can see on `pack builders suggest`. 

In the future, you can also open issues on the https://github.com/buildpacks/pack/ repo.

**Reply by ubermensch37** [reply]
Thank you for answering, that makes now a lot of sense. 

My best regards
